File: backend/routes/communities.js
const express = require('express');
const router = express.Router();
const Community = require('../models/Community');
const auth = require('../middleware/auth');

// GET /api/communities
router.get('/', async (req, res) => {
  const communities = await Community.find();
  res.json(communities);
});

// POST /api/communities (protected)
router.post('/', auth, async (req, res) => {
  const { name, description, link, icon } = req.body;
  const community = new Community({ name, description, link, icon });
  await community.save();
  res.status(201).json(community);
});

module.exports = router; 

File: backend2/app/routes/communities.py
from fastapi import APIRouter, Depends
from typing import List, Dict, Any
from ..middleware.auth import get_current_user_optional

router = APIRouter()

@router.get("/")
async def get_communities():
    """Get all communities"""
    # Mock data for now - replace with actual database calls
    return [
        {"id": "1", "name": "Kolkata Photography", "description": "Photography enthusiasts", "icon": "FaCamera"},
        {"id": "2", "name": "Food Lovers", "description": "Explore Kolkata's cuisine", "icon": "FaUtensils"}
    ]

@router.post("/")
async def create_community(
    community_data: dict,
    current_user: Dict[Any, Any] = Depends(get_current_user_optional)
):
    """Create a new community"""
    return {"message": "Community created", "data": community_data}